apps/web/src/server/app/auth/auth.controller.js

```diff
@@ -97,7 +97,7 @@ export function authSignIn(request, response, next) {
  * @param {import('express').NextFunction} next express next function
  */
 export async function handleRedirect(request, response, next) {
-	if (request.query.state) {
+	try {
```

Comment:
The only other code that could benefit from this is `cryptoUtils`. I am not expecting that to fail, but having the state is mandatory.

Why did you change the if to a try?

Reply:
This was actually a bug fix. Sending a rogue state value like `x` to this endpoint was actually causing the express app to hang somehow (the test would time out).

Furthermore, it's not sufficient just to check that there is a state in the query, as parsing/decrypting this state also relies on `request.session.key`, which could be undefined. Given the multiple possibilities for failure, and an existing try/catch being there, I just moved it to the top level, to catch any issue.

Reply:
Not sure how one could ever send a rogue state? You mean from our code?

As for that key dependency all of it is encrypted in `authSignIn(request, response, next)` so in theory you should have everything you need.

What I am trying to say is that if I hit the redirect url directly, what happens with the controller if we don't check for the presence of state?

Also maybe state is not the best name for it as its specific to auth.

Reply:
This function can be hit directly from a urlL Like you say, if you hit the url directly, then anything can be sent as state. Even if azure hits the url, the session may not longer exist, which also causes this handler to throw.

We can revert to an `if` but then we also have to add an `if` to the session key, and session nonce. Basically, there's a number of points of failure if this endpoint is hit improperly (or even properly, if the session has disappeared), so a try catch seems safer?

Reply:
I see what you mean now. The crypto decrypt could fire a TypeError if its deps are missing or rubish. Not sure if that can ever happen but I agree now that is safer like this.